maxfw/core/utils.py

```diff
@@ -0,0 +1,60 @@
+#
+# Copyright 2018-2019 IBM Corp. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+from flask import abort
+from maxfw.utils.image_utils import ImageProcessor
+
+
+def redirect_errors_to_flask(func):
+    """
+    This decorator function will capture all Pythonic errors and return them as flask errors.
+
+    If you are looking to disable this functionality, please remove this decorator from the `apply_transforms()` module
+    under the ImageProcessor class.
+    """
+
+    def inner(*args, **kwargs):
+        try:
+            # run the function
+            return func(*args, **kwargs)
+        except ValueError as ve:
+            if 'pic should be 2 or 3 dimensional' in str(ve):
+                abort(400, "Invalid input dimensions, please ensure the input is a grayscale,",
+                      "RGB or RGBA image or a corresponding numpy array of 2 or 3 dimensions.")
+        except TypeError as te:
+            if 'bytes or ndarray' in str(te):
+                abort(400, "Invalid input format, please make sure the input is an image or numpy array")
```

Comment:
Similarly here, perhaps remove the numpy array reference?

Reply:
Fixed in 108c321.